--- Form/Type/Select2EntityType.php	(revision )
+++ Form/Type/Select2EntityType.php	(revision )
@@ -71,7 +71,7 @@
                 throw new \Exception('Unable to load class: '.$options['transformer']);
             }

-            $transformer = new $options['transformer']($this->em, $options['class']);
+            $transformer = new $options['transformer']($this->em, $options['class'], $options['text_property'], $options['primary_key']);

             if (!$transformer instanceof DataTransformerInterface) {
                 throw new \Exception(sprintf('The custom transformer %s must implement "Symfony\Component\Form\DataTransformerInterface"', get_class($transformer)));
